test/integration/progressesTasks.test.js

```diff
@@ -222,6 +222,34 @@ describe("Test Progress Updates API for Tasks", function () {
         });
     });
 
+    it("Returns the progress array for the task with userData object", function (done) {
+      chai
+        .request(app)
```

Comment:
Can we please write a test with an empty progress dataset to ensure the response structure is as expected?

Reply:
added please check